### PR TITLE
probably fixes emags only hacking lockers on harm intent

### PR DESCRIPTION
**Pull request by imsxz**
:cl:imsxz
fix: emags now hack lockers regardless of intent
/:cl:
please god tell me if theres a better way of doing this i couldnt figure one out it looks so awful aaaaaaaa

### CODE REVIEW

**Comment by XDTM** [comment]
emags already have NOBLUDGEON, so this should do nothing

**Comment by SpaceManiac** [comment]
emags don't override `GetID` so it should be returning `null` by default

**Comment by ExcessiveUseOfCobblestone** [comment]
default card is src

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -267,7 +267,7 @@
 						"<span class='italics'>You hear a ratchet.</span>")
 	else if(user.a_intent != INTENT_HARM)
 		var/item_is_id = W.GetID()
-		if(!item_is_id && !(W.item_flags & NOBLUDGEON))
+		if(!item_is_id && !(W.item_flags & NOBLUDGEON) || istype(W, /obj/item/card/emag))
 			return FALSE
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/closets.dm]: return FALSE means you hit the item, emag needs to return TRUE

**Comment by Bawhoppen** [comment]
Would you please test that the e-katana still works also? It normally only works to emag doors and nothing else, and can only do so on harm intent.

**Comment by AlexTheSysop42** [comment]
ekatana works (or worked) to emag lots of other things too, but it teleports you onto the tile and hits it as well

**Comment by Bawhoppen** [comment]
@AlexTheSysop42 Oh, was it changed how it works, or did I just not ever know it worked on some other stuff? (It never used to teleport also)

**Comment by ExcessiveUseOfCobblestone** [comment]
@imsxz 

If it's already set to `no_bludgeon` then no form of order of operations is going to fix the issue.

**Comment by optimumtact** [comment]
this is not the right fix

**Pull request closed without merging**